tests/test-firefox/test.firefox.js

```diff
@@ -480,29 +481,53 @@ describe('firefox', () => {
         });
     });
 
-    it('throws an error when the port is occupied', () => {
-      // TODO: add a retry for occupied ports.
-      // https://github.com/mozilla/web-ext/issues/283
-      const client = fake(RemoteFirefox.prototype);
-      const connectToFirefox = sinon.spy(() => Promise.resolve(client));
-      return findRemotePort({connectToFirefox})
-        .then(makeSureItFails())
-        .catch(onlyInstancesOf(WebExtError, (error) => {
-          assert.match(error.message, /Cannot listen on port/);
-          assert.equal(client.disconnect.called, true);
+    it('returns a port on first try', () => {
+      const connectToFirefox = sinon.spy(() => new Promise(
+        (resolve, reject) => {
+          reject(
+            new TCPConnectError('first call connection fails - port is free')
+          );
         }));
+      return findRemotePort({connectToFirefox})
+      .then((port) => {
+        assert.equal(connectToFirefox.callCount, 1);
+        assert.isNumber(port);
+      });
     });
 
-    it('re-throws unexpected connection errors', () => {
-      const connectToFirefox = sinon.spy(
-        () => Promise.reject(new Error('not a connection error')));
+    it('cancel discovery after too many fails', () => {
+      const client = fake(RemoteFirefox.prototype);
+      const connectToFirefox = sinon.spy(() => new Promise(
+        (resolve) => {
+          resolve(client);
+        }));
       return findRemotePort({connectToFirefox})
```

Comment:
This function will try for 10 times before giving up. This will slow down the test suite unnecessarily so what we do here instead is to limit the number of times it can try before giving up (just for testing purposes). You would call it like this:

    return findRemotePort({connectToFirefox, retriesCount: 2})

You would then need to adjust the signature of the function to accept a custom `retriesCount` value. With this change, the test would only try twice instead of 10 times but it would still cover the behavior we want to test, i.e. the function will give up after X times.